cmd/syncthing/gui.go

```diff
@@ -267,9 +268,15 @@ func (s *apiService) Serve() {
 	postRestMux.HandleFunc("/rest/system/resume", s.postSystemResume)          // device
 	postRestMux.HandleFunc("/rest/system/debug", s.postSystemDebug)            // [enable] [disable]
 
-	// Debug endpoints, not for general use
-	getRestMux.HandleFunc("/rest/debug/peerCompletion", s.getPeerCompletion)
-	getRestMux.HandleFunc("/rest/debug/httpmetrics", s.getSystemHTTPMetrics)
+	guiCfg := s.cfg.GUI()
+
+	if guiCfg.EnableDebugging {
+		// Debug endpoints, not for general use
+		getRestMux.HandleFunc("/rest/debug/peerCompletion", s.getPeerCompletion)
+		getRestMux.HandleFunc("/rest/debug/httpmetrics", s.getSystemHTTPMetrics)
+		getRestMux.HandleFunc("/rest/debug/cpuprof", s.getCPUProf) // duration
+		getRestMux.HandleFunc("/rest/debug/heapprof", s.getHeapProf)
+	}
```

Comment:
Should we always add these, and then just return 500s if the var is not true? Prevents need to restart.